cmd/kubernetes-admission-controller/validation/validator.go

```diff
@@ -0,0 +1,38 @@
+package validation
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/anchore/kubernetes-admission-controller/cmd/kubernetes-admission-controller/anchore"
+
+	"k8s.io/klog"
+)
+
+// Validator is a function that can be executed to produce a validation Result.
+type Validator func() Result
+
+// New returns a validator function with its internal state fully configured.
+// If a validator function cannot be created, New returns an error.
+func New(configuration Configuration, imageBackend anchore.ImageBackend, user anchore.Credential,
+	imageReference string) (Validator, error) {
+	switch mode := configuration.Mode; mode {
+	case PolicyGateMode:
+		return func() Result {
+			return policy(imageBackend, user, imageReference, configuration.PolicyReference.PolicyBundleId)
+		}, nil
+
+	case AnalysisGateMode:
+		return func() Result {
+			return analysis(imageBackend, user, imageReference)
+		}, nil
+
+	case BreakGlassMode:
+		return breakGlass, nil
+
+	default:
+		message := fmt.Sprintf("got unexpected mode value %q for matching selector. failing on error.", mode)
+		klog.Error(message)
```

Comment:
nit: logging the error at the right level seems to be up to the caller (the message specifically calls out "failing on error"). Something like:
```go
if err != nil {
   klog.Error(err)
   return validation.Result{...}
}
```
up around the caller (https://github.com/anchore/kubernetes-admission-controller/pull/40/files#diff-6fe3f359b366ce5d6a1a2cd59a4ae02c4f35cefa7c2fcc250e8a90a768bbf60bR178)

Reply:
@wagoodman Good catch! This is inconsistent; and, the `validation` package shouldn't report messages like "failing on error", because ultimately deciding failure is the `admission` package's responsibility. This has been adjusted to be consistent with other error handling in the project.

Let me know what you think: https://github.com/anchore/kubernetes-admission-controller/pull/40/commits/49afc05b20aca5faecc2b4ea5798f641db05efa0